From 691b6ea8e100fc55f0ff70a82511722733113a07 Mon Sep 17 00:00:00 2001
From: Tor Lillqvist <tml@iki.fi>
Date: Tue, 28 May 2013 05:43:57 +0000
Subject: Make the SAL_STREAM thing compile with clang++ -std=c++11 -stdlib=libc++

Avoid error: dynamic_cast from rvalue to reference type
'::std::ostringstream &'.

Change-Id: If5ba1ee79616e81b12f9db67c61b4f085b30a08e
---
(limited to 'include/sal/log.hxx')

diff --git a/include/sal/log.hxx b/include/sal/log.hxx
index 80721d0..ca3bb5e 100644
--- a/include/sal/log.hxx
+++ b/include/sal/log.hxx
@@ -158,8 +158,7 @@ inline char const * unwrapStream(SAL_UNUSED_PARAMETER StreamIgnore const &) {
     @since LibreOffice 3.5
 */
 #define SAL_STREAM(stream) \
-    (dynamic_cast< ::std::ostringstream & >(::std::ostringstream() << stream). \
-     str())
+    (::std::ostringstream() << stream).str()
 
 /**
     @page sal_log Basic logging functionality.
--
cgit v0.9.0.2-2-gbebe
